lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -517,16 +548,38 @@ class ModuleEmitter : public EmitterBase {
     return entry->getKey();
   }
 
+  /// Given a module name \p moduleName, return the updated name if it has been
+  /// previously renamed in the table moduleNameTable, else call
+  /// resolveKeywordConflict, to get a new name in case of conflict with
+  /// keywords.
+  StringRef getModuleName(StringRef moduleName) {
+    if (!moduleNameTable.count(moduleName)) {
+      moduleNameTable[moduleName] = &*resolveKeywordConflict(
+          moduleName, usedModuleNames, nextGeneratedNameID);
+    }
+    return moduleNameTable[moduleName]->getKey();
+  }
+
 public:
   /// nameTable keeps track of mappings from Value's and operations (for
   /// instances) to their string table entry.
   llvm::DenseMap<ValueOrOp, llvm::StringMapEntry<llvm::NoneType> *> nameTable;
 
+  /// moduleNameTable keeps track of mappings from Module names to updated names
+  /// to resolve keyword conflicts.
+  llvm::DenseMap<StringRef, llvm::StringMapEntry<llvm::NoneType> *>
+      moduleNameTable;
```

Comment:
Another small thing, but I'd recommend changing the key of this (and the argument of `getModuleName`) to be a `StringAttr`.  `StringAttr` is a single pointer in size and is already uniqued.

Reply:
Modified it to ```StringAttr```